### ISSUE TITLE
serverless deploy function -f not packaging correctly.

### ISSUE THREAD

**Issue by selected-pixel-jameson** [serverless/serverless#3625]
I'm having issues when running.

`serverless deploy function -f  <function>` 

I haven't determined what version this was introduced on, but if I run a serverless deploy everything seems to be deployed correctly. If I deploy a single function I received the following error.

```
{
  "errorMessage": "Class not found: com.xxxxxxx.user.RequestHandler",
  "errorType": "class java.lang.ClassNotFoundException"
}
```

This is a Maven project running Java 1.8 and Serverless 1.13.2. 

On thing that I notice is the package size is very different between Serverless 1.10 and Serverless 1.13.2. It goes from 24.27mb -> 30.25mb

Also in my serverless.yml file I'm explicitly referencing the the package artifact for every function. This was needed awhile back to get the individual functions to deploy with a maven package. Is it maybe not needed any longer and that is my underlying issue? 

```
package:
      artifact: target/xxxxxxxx-${self:custom.stage}.jar
```

**Comment by selected-pixel-jameson** [comment]
I reverted back to serverless 1.10 and this is no longer a problem. So it's something that happened between there that has caused my issue.

**Comment by jjfine** [comment]
I'm seeing this issue as well. only differences are its Gradle project using sls 1.13.0

**Comment by javierros** [comment]
Hi, I've seen a modification in the file serverless/lib/plugins/package/lib/packageService.js (https://github.com/serverless/serverless/blob/21b6037d05020a6e4a6ec54cc77dcf5899f4f8d0/lib/plugins/package/lib/packageService.js)

In the function packageFunction we should have this piece of code:

if (funcPackageConfig.artifact) {
      functionObject.artifact = funcPackageConfig.artifact;
      return BbPromise.resolve(funcPackageConfig.artifact);
}

It used to be in the version 1.10, but it was deleted, can it be added again?

**Comment by pmuens** [comment]
@javierros thanks (again) for looking that up 👍 Looks reasonable 🤔 

/cc @eahefnawy 🔝

**Comment by selected-pixel-jameson** [comment]
Has this been implemented yet??

**Comment by pmuens** [comment]
> Has this been implemented yet??

There has been no PR which targets this yet. Adding it to the next milestone to track that.

/cc @eahefnawy

**Comment by pmuens** [comment]
Just a quick update on this one. @darkowlzz PRed a fix in #3856 which should resolve this bug.
